zio-http/src/main/scala/zio/http/package.scala

```diff
@@ -1,13 +1,16 @@
 package zio
 
+import zio.http.socket.WebSocketChannelEvent
 import zio.stacktracer.TracingImplicits.disableAutoTrace // scalafix:ok;
 
 package object http extends PathSyntax with RequestSyntax with RouteDecoderModule {
-  type HttpApp[-R, +E]                       = Http[R, E, Request, Response]
-  type UHttpApp                              = HttpApp[Any, Nothing]
-  type RHttpApp[-R]                          = HttpApp[R, Throwable]
-  type EHttpApp                              = HttpApp[Any, Throwable]
-  type UHttp[-A, +B]                         = Http[Any, Nothing, A, B]
+  type HttpApp[-R, +E] = Http[R, E, Request, Response]
+  type UHttpApp        = HttpApp[Any, Nothing]
+  type RHttpApp[-R]    = HttpApp[R, Throwable]
+  type EHttpApp        = HttpApp[Any, Throwable]
+  type UHttp[-A, +B]   = Http[Any, Nothing, A, B]
+  type HttpSocket      = Http[Any, Throwable, WebSocketChannelEvent, Unit]
```

Comment:
This addition probably warrants the most scrutiny. I found it useful, but I know expanding the API here should not be done lightly.

Reply:
I would probably not introduce this just yet given the looming changes for web socket. As for the other ones, I'm not personally a fan of the benefits (of less typing) given the costs (of indirection) but don't have a huge preference either way.

Reply:
Removed